### ISSUE TITLE
ReGreet not using GTK or cursor theme

### ISSUE THREAD

**Issue by yazoink** [danth/stylix#690]
It's using the correct font and wallpaper but the GTK colors and cursor theme have not been applied.

**Comment by trueNAHO** [comment]
Cc: @Mikilio

**Comment by Mikilio** [comment]
Currently Regreet is only themed by copying the configuration for GTK and applying it to Regreet. If that is not working then it means there is a good reason to create a mustache file for regreet only. If I am to do that I'd like to do it properly, so it takes me more than just a few minutes.

**Comment by Mikilio** [comment]
Waiting for https://github.com/NixOS/nixpkgs/issues/367908 to close so I can use demo mode instead of rebooting to see a css.

**Comment by Mikilio** [comment]
I can't reproduce the issue. So I guess I need a bit more guidance on the error you're experiencing.
@yazoink, can you maybe send me the stylix related options you are using?

**Comment by yazoink** [comment]
[Here](https://github.com/yazoink/nixos/blob/main/modules/nixos/features/theme/default.nix) and [here](https://github.com/yazoink/nixos/blob/main/modules/nixos/features/theme/default.nix) are my stylix options and [this](https://github.com/yazoink/nixos/blob/main/modules/nixos/features/greetd/default.nix) is how I set up regreet.

**Comment by Mikilio** [comment]
A few info I'd like to provide in advance:
 - If you overwrite `programs.regreet.extraCss` with `mkForce` you simply overwrite the entire css that stylix defines with your own, which I believe is not what you're trying to do if you want to extend the config writing stuff into `stylix.gtk.extraCss` is the way to go though I admit that could be improved.
 - regarding your regreet and greetd config something like the following is enough to setup greetd with regreet while having it styled only by stylix:
 ```
  programs.regreet = {
    enable = true;
  };
 ```
 - The font used for UI is sansSerif because of the styling guide and because serifs are really ugly on UI.
 
 To follow up I recommend removing the clutter from your configuration and tweaking `stylix.fonts.sansSerif` to see if that changes your font.
 
 If it does not change your font I believe you will be forced to debug this using `nix repl` after loading your config with `:lf .` you can check the values of your config after evaluation with `nixosConfigurations.cyberia.config.<ENTER-YOUR-KEY>`.

**Comment by yazoink** [comment]
It's not being themed even without the `programs.regreet.extraCss`. This is what I have currently:
```
 programs.regreet = {
   enable = true;
 };
 services.greetd = {
   enable = true;
   settings = {
     default_session = {
       command = "${lib.getExe pkgs.cage} -s -- ${lib.getExe config.programs.regreet.package}";
       user = "greeter";
     };
   };
 };
```

**Comment by Mikilio** [comment]
Thank you for your patience! I was able to find the error, it seems my regreet was being styled by GTK through importing correct environment variables, not through extraCss.

As a tip for your config, I would recommend removing everything under `services.greetd` as that is not needed and is actually missing parts from the default that is already quite minimal.

**Comment by Mikilio** [comment]
Even without the above change, Regreet will be themed with the global GTK settings, if use the default value for `services.greetd.settings.default_session.command` that is set when `programs.regreet.enable = true`.

It will set up the dbus environment which will firstly prevent a [known issue](https://github.com/swaywm/sway/wiki#gtk-applications-take-20-seconds-to-start) affecting ReGreet.
Since in nixpkgs this is activated by default, I think it is sensible to assume that if deliberately turned it off a user is aware of its implications.

I could still not replicate your issue with fonts.
Could you report what happens when you change sansSerif fonts in stylix?

**Comment by yazoink** [comment]
It's using the correct cursor theme and font now but the GTK theme still isn't working. This is my current config for regreet/greetd:
```nix
programs.regreet = {
  enable = true;
};
```
Am I missing something?

**Comment by Mikilio** [comment]
To rule out that it is some light or dark mode issue, it would be helpful if you could try the PR. It should also get merged soon.

**Comment by niLPotential** [comment]
Is this issue still open? Because I am having a similar problem. For me, only the wallpaper is applied, but not the GTK theme.
The GTK theme I am using is generated by stylix and is not global.
Therefore ReGreet uses the default `adw-gtk3` theme at `/run/current-system/sw/share/themes`, since it is unaware of user specific themes.
Copying the files to said system directories would solve this, I believe.
But the global theme configs need to be generated on build as well, and I am uncertain how to implement it.

**Comment by danth** [comment]
Yes, installing the NixOS theme as the system wide default (while still being able to override it with Home Manager for individual users) would be ideal.

**Comment by Mikilio** [comment]
I don't understand how this should be possible, to be honest, as a "greeter" should obviously not run as any normal user. After all, it runs, so a user can actually log in. Should it just sneak a peek at some user's GTK config and apply it? Does not sound very reproducible.

**Comment by Mikilio** [comment]
Since I had the privilege of configuring a system not using catppuccin it became apparent to me that this issue is actually still real.
It seems that regreet doesn't care about dconf.
